extensions-contrib/scan-query/src/main/java/io/druid/query/scan/ScanQueryEngine.java

```diff
@@ -173,10 +175,12 @@ public ScanResultValue next()
                                 ScanQueryRunnerFactory.CTX_COUNT,
                                 (long) responseContext.get(ScanQueryRunnerFactory.CTX_COUNT) + (offset - lastOffset)
                             );
-                            responseContext.put(
-                                ScanQueryRunnerFactory.CTX_TIMEOUT_AT,
-                                timeoutAt - (System.currentTimeMillis() - start)
-                            );
+                            if (hasTimeout) {
+                              responseContext.put(
```

Comment:
Thanks for letting me know. This behavior looks weird, so I'd like to fix it in this pr if it doesn't require so large change. Since I don't know exactly how the responseContext is supposed to be used, let me make sure what the correct solution is. Is it ok to just move CTX_TIMEOUT_AT to QueryContext instead of ResponseContext?